packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+    accounts = new Accounts(true);
+    lockedGold = new LockedGold(true);
+    election = new MockElection();
+    validators = new MockValidators();
+    governance = new MockGovernance();
+    stableToken = new MockStableToken();
+
+    registry.setAddressFor("Accounts", address(accounts));
+    registry.setAddressFor("Election", address(election));
+    registry.setAddressFor("GoldToken", address(goldToken));
+    registry.setAddressFor("Governance", address(governance));
+    registry.setAddressFor("LockedGold", address(lockedGold));
+    registry.setAddressFor("Validators", address(validators));
+    registry.setAddressFor("StableToken", address(stableToken));
+    lockedGold.initialize(address(registry), unlockingPeriod);
+    accounts.createAccount();
+  }
+
+  function getParsedSignatureOfAddress(address _address, uint256 privateKey)
+    public
+    pure
+    returns (uint8, bytes32, bytes32)
+  {
+    bytes32 addressHash = keccak256(abi.encodePacked(_address));
+    bytes32 prefixedHash = ECDSA.toEthSignedMessageHash(addressHash);
+    return vm.sign(privateKey, prefixedHash);
+  }
+
+  function authorizeVoteSigner(address _delegator, uint256 _signerPK) public {
+    (uint8 vVoting, bytes32 rVoting, bytes32 sVoting) = getParsedSignatureOfAddress(
+      _delegator,
+      _signerPK
+    );
+
+    address signerAddress = vm.addr(_signerPK);
+
+    vm.prank(_delegator);
+    accounts.authorizeVoteSigner(signerAddress, vVoting, rVoting, sVoting);
+
+    vm.prank(signerAddress);
+    accounts.completeSignerAuthorization(
+      _delegator,
+      keccak256(abi.encodePacked("celo.org/core/vote"))
+    );
+  }
+
+  function createAndAssertDelegatorDelegateeSigners(
+    address _delegator,
+    address _delegatee,
+    uint256 _delegatorSignerPK,
+    uint256 _delegateeSignerPK
+  ) public {
+    if (_delegator != address(0)) {
+      authorizeVoteSigner(_delegator, _delegatorSignerPK);
+      assertFalse(_delegator == vm.addr(_delegatorSignerPK));
+      assertEq(accounts.voteSignerToAccount(vm.addr(_delegatorSignerPK)), _delegator);
+    }
+    if (_delegatee != address(0)) {
+      authorizeVoteSigner(_delegatee, _delegateeSignerPK);
+      assertFalse(_delegatee == vm.addr(_delegateeSignerPK));
+      assertEq(accounts.voteSignerToAccount(vm.addr(_delegateeSignerPK)), _delegatee);
+    }
+  }
+
+  function assertDelegatorDelegateeAmounts(
+    address _delegator,
+    address delegatee,
+    uint256 percent,
+    uint256 amount
+  ) public {
+    (uint256 fraction, uint256 currentAmount) = lockedGold.getDelegatorDelegateeInfo(
+      _delegator,
+      delegatee
+    );
+    assertEq(FixidityLib.wrap(fraction * 100).fromFixed(), percent, "fraction incorrect");
+    assertEq(currentAmount, amount, "amount incorrect");
+  }
+}
+
+contract Initialize is LockedGoldFoundryTest {
+  function setUp() public {
+    super.setUp();
+  }
+
+  function test_ShouldSetOwner() public {
+    assertEq(lockedGold.owner(), address(this));
```

Comment:
To mock owner I usually define `address caller = address(this);` in the contract, this could even be added to this file I created: https://github.com/celo-org/celo-monorepo/pull/10773/files#diff-f7395554b886736da22e519de74623b3087a6c9239a351df77c8586cc6841c07

Reply:
I added caller variable, but I dont believe we can add it as constant since it is dynamic value